cannon/mipsevm/evm_test.go

```diff
@@ -281,30 +280,14 @@ func TestClaimEVM(t *testing.T) {
 
 		stepWitness, err := us.Step(true)
 		require.NoError(t, err)
-		input := stepWitness.EncodeStepInput()
-		startingGas := uint64(30_000_000)
-
-		// we take a snapshot so we can clean up the state, and isolate the logs of this instruction run.
-		snap := env.StateDB.Snapshot()
-
-		// prepare pre-image oracle data, if any
-		if stepWitness.HasPreimage() {
-			poInput, err := stepWitness.EncodePreimageOracleInput()
-			require.NoError(t, err, "encode preimage oracle input")
-			_, leftOverGas, err := env.Call(vm.AccountRef(addrs.Sender), addrs.Oracle, poInput, startingGas, big.NewInt(0))
-			require.NoErrorf(t, err, "evm should not fail, took %d gas", startingGas-leftOverGas)
-		}
 
-		ret, leftOverGas, err := env.Call(vm.AccountRef(addrs.Sender), addrs.MIPS, input, startingGas, big.NewInt(0))
-		require.NoErrorf(t, err, "evm should not fail, took %d gas", startingGas-leftOverGas)
-		require.Len(t, ret, 32, "expecting 32-byte state hash")
-		// remember state hash, to check it against state
-		postHash := common.Hash(*(*[32]byte)(ret))
-		logs := evmState.Logs()
-		require.Equal(t, 1, len(logs), "expecting a log with post-state")
-		evmPost := logs[0].Data
-		require.Equal(t, crypto.Keccak256Hash(evmPost), postHash, "logged state must be accurate")
-		env.StateDB.RevertToSnapshot(snap)
+		evm := NewMIPSEVM(contracts, addrs)
+		evm.SetTracer(tracer)
+		evmPost := evm.Step(t, stepWitness)
+
+		uniPost := us.state.EncodeWitness()
```

Comment:
All the `uniPost` occurrences may be worth renaming to e.g. `goPost`, since we don't use unicorn anymore.